### ISSUE TITLE
Events: Migrate from CSR to SSR

### ISSUE THREAD

**Issue by EthanThatOneKid** [acmcsufoss/acmcsuf.com#287]
**tl;dr**: Refactor how `acmcsuf.com/events` renders the event list. Make sure the HTML for the cards is generated on the server, instead of retroactively on the client's side.

Currently, `acmcsuf.com/events` fetches the event data from `acmcsuf.com/events.json` on the client after the page loads. This forced us to display a "loading..." dialog during the moment that it takes to load the event data. For our purposes, it is better to bake the event data into the HTML before it is sent to the client so that the user does not have to wait for the event data to be fetched. 

To resolve this bug, the event list items must be visible from `acmcsuf.com/events`'s page source.
